recipe/meta.yaml

```diff
@@ -179,6 +179,7 @@ outputs:
         - ld_impl_{{ target_platform }} >=2.36.1  # [linux]
         - libnsl  # [linux]
         - libuuid  # [linux]
+        - libxcrypt  # [linux]
```

Comment:
Do we need to add this to `requirements/run` as well? Or should `libxcrypt` have `run_exports`?

Asking because of issues like this one: https://github.com/conda-forge/ray-packages-feedstock/pull/150#issuecomment-1960928574

cc @h-vetinari

Reply:
That's not the question though

The question is why we are still needing to work around this in packages:

* https://github.com/conda-forge/ray-packages-feedstock/pull/150#issuecomment-1960928574
* https://github.com/conda-forge/torch-scatter-feedstock/pull/25#discussion_r1523989702

Reply:
> That's not the question though

That was _literally_ the question you asked.

> The question is why we are still needing to work around this in packages:

If you have downstream builds that do not find headers/libraries then the first thing to check is whether those builds respect the `CFLAGS`/`CXXFLAGS`/`LDFLAGS` we provide.

Reply:
Hey Marcel, I'm sorry if I've said something that offended you. That was not my intent 😞

Agree that is a good thing to check 👍

Am mentioning this as we are actually seeing a few cases where users were able to build Python extensions ok (so presumably they got Python headers without issues before). However when it comes to `crypt.h` or `-lcrypt` they are running into issues:

* https://github.com/conda-forge/torchmd-net-feedstock/pull/21#issuecomment-1956007199
* https://github.com/conda-forge/libtiledb-sql-py-feedstock/pull/17#issuecomment-1954224399

While it certainly could be flag issues, I'm curious how they are able to find other things like `Python.h`, but not `crypt.h`. Would be interested in hearing your insights

Reply:
> While it certainly could be flag issues, I'm curious how they are able to find other things like Python.h, but not crypt.h. Would be interested in hearing your insights

Can you check whether Python.h and crypt.h are in the same location? Then you should have your answer.

Reply:
> Hey Marcel, I'm sorry if I've said something that offended you. That was not my intent 😞

Oh, don't worry no offense taken.
(In case my use of "literally" came over as harsh: I meant its literal ( :laughing: ) sense -- not in the often encountered hyperbolical misuse!)

The issue is just that such things grab the attention of some(/multiple) of our most valued members (maintainers of `python` package/core members) and as such tap into important/scarce resources of ours -- which I'd like us not to do if it can be avoided by doing some quick checks beforehand (which I'd kind of expect from seasoned members like yourself).

----
As Isuru hinted at, `Python.h` is not located directly under the main `include` directory; as such builds which only include paths to `Python.h`'s directory won't find other headers.
(The reason that only 3.8 builds are affected is just that later versions don't include `config.h` in `Python.h`.)
Mind you, it is actually a good thing that these builds failed then; it gives you an indication that other flags (e.g., CPU arch) that we'd want to use for our packages are left out, too.

Reply:
Ok fair enough

My point is more that we are seeing recurring issues with a security fix where maintainers are frequently throwing up their hands and linking to issue ( https://github.com/conda-forge/linux-sysroot-feedstock/issues/52 ) with various hacks they had to implement. So am concerned something isn't work right (am not claiming to know what it is yet and it may not just be one thing), but the nature of these problems does merit more attention

> (The reason that only 3.8 builds are affected is just that later versions don't include config.h in Python.h.)
Mind you, it is actually a good thing that these builds failed then; it gives you an indication that other flags (e.g., CPU arch) that we'd want to use for our packages are left out, too.

Ah ok! This is the sort of insight I was hoping for. Thanks Marcel! 🙏